crates/autopilot/src/on_settlement_event_updater.rs

```diff
@@ -199,7 +199,7 @@ impl Inner {
 
             tracing::info!(
                 "settlement object {:?}",
-                settlement.map(|settlement| (settlement.observation(), settlement.score()))
```

Comment:
Why not keep this line and turn invalid score computations into an `ScoreMismatch` error that has both scores and gets printed?

Reply:
removed public function `score` in the meantime. I reverted the `score < promised.solution` to return error, although i am sceptic about it, as it won't catch potential bugs in score calculation when score is much higher than promised.solution -> so I added temporary log to observe this.